File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n=nums.size();

        sort(nums.begin(),nums.end());
        int f=1,ans=nums[0];
        for (int i=1;i<n;i++){
            if(nums[i]==nums[i-1]){
                f++;
            }else{
                f=1;
                ans=nums[i];
            }
            if(f>(n/2)){
                return ans;
            }
        }
        return ans;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int f=0,ans=0;
        for (int i=0;i<nums.length;i++){
            if(f==0) ans=nums[i];
            if(ans==nums[i]) f++;
            else f--;
        }
        return ans;
    }
}